pkg/reloader/collector.go

```diff
@@ -93,24 +94,43 @@ func (c *Controller) collectWorkloadSecrets(workload workload, template corev1.P
 	collectorLogger.Infof("Collected secrets from %s %s/%s", workload.kind, workload.namespace, workload.name)
 }
 
-func collectSecretsFromContainerEnvVars(containers []corev1.Container) []string {
+func collectSecrets(template corev1.PodTemplateSpec) []string {
+	containers := []corev1.Container{}
+	containers = append(containers, template.Spec.Containers...)
+	containers = append(containers, template.Spec.InitContainers...)
+
+	vaultSecretPaths := []string{}
+	collectSecretsFromContainerEnvVars(containers, &vaultSecretPaths)
+	collectSecretsFromVaultEnvFromPathAnnotation(template.GetAnnotations()[SecretPathsAnnotation], &vaultSecretPaths)
+
+	// Remove duplicates
+	slices.Sort(vaultSecretPaths)
+	return slices.Compact(vaultSecretPaths)
+}
+
+func collectSecretsFromContainerEnvVars(containers []corev1.Container, vaultSecretPaths *[]string) {
 	// Iterate through all environment variables and extract secrets
-	var vaultSecrets []string
 	for _, container := range containers {
 		for _, env := range container.Env {
 			// Skip if env var does not contain a vault secret or is a secret with pinned version
 			if hasVaultPrefix(env.Value) && unversionedSecretValue(env.Value) {
 				secret := regexp.MustCompile(`vault:(.*?)#`).FindStringSubmatch(env.Value)[1]
 				if secret != "" {
-					vaultSecrets = append(vaultSecrets, secret)
+					*vaultSecretPaths = append(*vaultSecretPaths, secret)
 				}
 			}
 		}
 	}
+}
 
-	// Remove duplicates
-	slices.Sort(vaultSecrets)
-	return slices.Compact(vaultSecrets)
+func collectSecretsFromVaultEnvFromPathAnnotation(secretPaths string, vaultSecretPaths *[]string) {
```

Comment:
`collectSecretsFromAnnotations` and pass down annotation data rather than secretPaths to ensure that all annotation-related collection logic is actually implemented here

Reply:
There is only one annotation exists to collect secret paths from though, but I'm fine with it either way